external/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaTestUtils.scala

```diff
@@ -485,3 +504,22 @@ class KafkaTestUtils(withBrokerProps: Map[String, Object] = Map.empty) extends L
   }
 }
 
+object KafkaTestUtils {
+
+  def assertEqual(lhs: (String, Array[(String, Array[Byte])]),
+                  rhs: (String, Array[(String, Array[Byte])])): Unit = {
+    assert(lhs._1 == rhs._1)
+    assert(lhs._2.size == rhs._2.size)
+    (0 until lhs._2.size) foreach { i =>
+      assert(lhs._2(i)._1 == rhs._2(i)._1)
+      assert(lhs._2(i)._2.deep == rhs._2(i)._2.deep)
+    }
+  }
+
+  def assertEqual(lhs: Array[(String, Array[(String, Array[Byte])])],
```

Comment:
I think scalatest will understand tuple comparison already. It should be able to compare arrays too. Check its methods to see how to do deep comparisons. You may not need more utility methods

Reply:
I tried it, but it throws an error; One alternative I think is defining `KafkaTest` trait like `AnalysisTest` which extends `SparkFunSuite` and make `KafkaDataConsumerSuite` extend it like `SparkSqlParserSuite` extends `AnalysisTest.` How about this plan?